python/pyarrow/table.pxi

```diff
@@ -2882,24 +2882,27 @@ cdef class Table(_PandasConvertible):
 
         return pyarrow_wrap_table(result)
 
-    def filter(self, mask, object null_selection_behavior="drop"):
+    def filter(self, mask_or_expr, object null_selection_behavior="drop"):
         """
         Select rows from the table.
 
-        See :func:`pyarrow.compute.filter` for full usage.
+        The Table can be filtered based on a mask, which will be passed to
+        :func:`pyarrow.compute.filter` to perform the filtering, or it can
+        be filtered through a boolean :class:`.Expression`
 
         Parameters
         ----------
-        mask : Array or array-like
-            The boolean mask to filter the table with.
+        mask_or_expr : Array or array-like or .Expression
+            The boolean mask or the :class:`.Expression` to filter the table with.
         null_selection_behavior
-            How nulls in the mask should be handled.
+            How nulls in the mask should be handled, does nothing if
+            an :class:`.Expression` is used.
```

Comment:
This is not possible to pass through to the filter node?

Reply:
Not in any way that I can see, the filter node has a pretty straightforward constructor: 
`explicit FilterNodeOptions(Expression filter_expression, bool async_mode = true)`, it only accepts an expression.

I think that if you care about special handling nulls, you probably want to build an expression that evaluates as you wish for nulls

Reply:
> I think that if you care about special handling nulls, you probably want to build an expression that evaluates as you wish for nulls

I don't think is possible to get the "emit null" behaviour by changing the expression (for dropping/keeping, you can explicitly fill the null with False/True, but for preserving the row as null, that's only possible through this option). I suppose that is a good reason this is an option of the filter kernel and not eg comparison kernels. 

Anyway, this is not that important given that the "drop" behaviour is the default for both (and is the typical behaviour you want, I think), but this might be something to open a JIRA for to add `FilterOptions` to the `FilterNodeOptions` (cc @westonpace would that make sense?)

Reply:
Uhm, not sure I follow, why you can't use an expression?
Given
```
>>> pa.table({"rows": [1, 2, 3, None, 5, 6]})
pyarrow.Table
rows: int64
----
rows: [[1,2,3,null,5,6]]
```

If I want to drop the nulls, I do
```
>>> t.filter(pc.field("rows") < 5)
pyarrow.Table
rows: int64
----
rows: [[1,2,3]]
```

If instead I want to keep the nulls, I do
```
>>> t.filter((pc.field("rows") < 5) | (pc.field("rows").is_null()))
pyarrow.Table
rows: int64
----
rows: [[1,2,3,null]]
```

Reply:
The option is about introducing nulls in the output data where the mask is null, not about preserving nulls from the input data. So for preserving nulls in the input, you can change your expression. But for introducing nulls, I don't think that is possible.

Reply:
Using your example table:

```
In [29]: t.filter(pa.array([True, None, True, False, False, False]))
Out[29]: 
pyarrow.Table
rows: int64
----
rows: [[1,3]]
```

vs

```
In [33]: t.filter(pa.array([True, None, True, False, False, False]), null_selection_behavior="emit_null")
Out[33]: 
pyarrow.Table
rows: int64
----
rows: [[1,null,3]]
```

The null is in a place where the original data had a "2"